GoogleForCreators/web-stories-wp
# Average background media color for page background

## Summary
Gets the average color of the background media and displays it under the background media as the page background color.
If there is no background media, the panel for Page background color is displayed.


## Relevant Technical Choices
Base color is assigned to each element upon insertion, so it can just be used later.
There are exceptions -- existing media elements and media uploaded by dragging to the editor from somewhere else, for those there’s a workaround that creates a small image out of the screen, gets the average color from there, and assigns the base color together with assigning the background.
The PR changes a bit the logic of how `combineElements` works since the baseColor has to be assigned at the same time to ensure having only one history entry for the action.


## To-do

- [x] Karma



## User-facing changes
- When there's background media, page background color panel is hidden
- In case of background media, the average color of the image/video poster is used as the background color.


## Testing Instructions
1. Add a Page
2. Verify it has the Page background color panel available.
3. Drag an image from library to be the background instead.
4. Verify the panel for Page background color is not visible anymore
5. Verify that the page background color in the front-end is displayed based on the image -- e.g. if you used a pink image as the background media, the page background should have pink color. This can be tested out by disabling cache and setting the mode to slow internet so that while the image is loaded the background color can be seen. Alternatively, the color can be checked from the source code, please ask is there's trouble finding the correct place.
6. Verify that the page background is assigned correctly also in these cases:
- Dragging and dropping existing media to background (both image/video)
- Assigning an existing media as background from the Design Panel (both image/video).



---



Fixes #3638
